packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished([txStatus, txStatus]);
+            sinon.assert.calledTwice(updateSpy);
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector with a TxStatus result once deactivated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            clock.tick(25);
+            transactionStatisticsCollector.deactivate();
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            sinon.assert.calledOnce(updateSpy);
+            transactionStatisticsCollector.getRoundFinishTime().should.equal(75);
+        });
+
+    });
+
+    describe('Updating a TransactionStatisticsCollector with a TxStatus result', () => {
+
+        it('should update successful transactions', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(50);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForSuccessful().should.equal(50);
```

Comment:
Maybe we should also assert that the corresponding "failed" metrics are not updated.